lib/GameConstants.h

```diff
@@ -745,6 +745,27 @@ DLL_LINKAGE std::ostream & operator<<(std::ostream & os, const ETerrainType acti
 
 ID_LIKE_OPERATORS_DECLS(ETerrainType, ETerrainType::EETerrainType)
 
+class DLL_LINKAGE EPathfindingLayer
+{
+public:
+	enum EEPathfindingLayer
+	{
+		WRONG = -2, AUTO = -1, LAND = 0, SAIL = 1, WATER, AIR, NUM_LAYERS
+	};
+
+	EPathfindingLayer(EEPathfindingLayer _num = WRONG) : num(_num)
+	{}
+
+	ID_LIKE_CLASS_COMMON(EPathfindingLayer, EEPathfindingLayer)

```

Comment:
(outside of scope of this pull)
I don't remember exact reason for creation of this ID_LIKE_CLASS_COMMON system. I think it was due to silent convertions of integers from/to enums causing issues with our ID's. With C++11 this is no longer a problem due to enum class and possibility to "inherit" enum from integers of any size.

Perhaps deprecating this code is a good idea (unless there was something else I can't remember right now)